### PR TITLE
Various improvements to the spec_helper

### PR DESCRIPTION
**Pull request by ekohl**
Otherwise it won't be used

### CODE REVIEW

File: moduleroot/spec/spec_helper.rb.erb [hunk 1]
```diff
@@ -1,3 +1,8 @@
+<%- if @configs['mock_with'] -%>
+RSpec.configure do |c|
+  c.mock_with <%= @configs['mock_with'] %>
```
Comment (bastelfreak) [inline moduleroot/spec/spec_helper.rb.erb]: this block is now duplicate. We probably can remove the other one. This need to be tested within one module

Comment (ekohl) [reply moduleroot/spec/spec_helper.rb.erb]: I just verified this on puppet-mongodb (in my local branch) that the later one is indeed redundant and it must be done here at the top.

**Comment by ekohl** [comment]
Updated to include various updates. See the individual commits for details.

**Comment by bastelfreak** [comment]
can you please run this against one module and link the PR to this one?

**Comment by ekohl** [comment]
https://github.com/voxpupuli/puppet-zabbix/pull/549

**Comment by ekohl** [comment]
https://github.com/voxpupuli/puppet-nginx/pull/1264 as well

**Pull request merged**